### ISSUE TITLE
Create a visualization of mutations relative to the protein sequence, its domains, mutation hotspots, etc.

### ISSUE THREAD

**Issue by malachig** [genome/civic-client#66]
Since we have mutations that are point sites or are more generally in certain domains or other large features of the protein.  For example, MD Anderson creates these for each gene and they are customized when you click on each mutation:
![screen shot 2015-03-04 at 3 13 17 pm](https://cloud.githubusercontent.com/assets/856728/6493587/4da6134c-c281-11e4-8e24-154b73b74c49.png)

**Comment by jmcmichael** [comment]
At some point I can definitely create a d3 visualization that can generate these in the client, I just need to know what data I will be getting to generate the figure and how it corresponds to the MD Anderson example.

**Comment by malachig** [comment]
Yeah, I think we should keep this on the wishlist for now.  There is actually a somewhat complex mix of data being represented here.  From left to right:

- A chromosome ideogram image.  The position of each cytoband (giemsa staining data) is indicated.
- The chromosome coordinate of the current gene is indicated on this image
- Next the mRNA molecule is depicted.  Somehow a single reference transcript has been chosen for each gene.  Each exon is enumerated and the relative size of each is depicted.  The start and end position of the protein open reading frame are indicated. 
-  Next the protein sequence is depicted with the relative position of protein domains (these appear to have been manually curated/simplified).   These are labelled with amino acid coordinates.
- Finally the position of the current variant is indicated on the protein sequence.
- One could also imagine a display that showed the landscape of all mutations on this figure in a lolliplot style.  

The raw material for most of the data needed could be obtained from the Ensembl API (or similar).  The variant landscape could come from Cosmic, the ICGC data portal, TCGA, etc.

Zach has been working on a similar visualization concept in R using R modules (Biomart?) to get the raw data and ggplot2 to create the image.  If we decide to tackle this at some point we should probably get together a group and brainstorm on some of the specifics.

**Comment by kkrysiak** [comment]
The GenVisR (Zach's R modules) protein visualization continues to evolve. https://github.com/griffithlab/GenVisR

Alternatively, in many cases we have all the fields to generate a lolliplot via cBioPortal. http://www.cbioportal.org/mutation_mapper.jsp The source code is also available on GitHub under a GNU Affero General Public License V3 https://github.com/cBioPortal/cbioportal

**Comment by kkrysiak** [comment]
Consider embedding protein paint. https://docs.google.com/document/d/1KNx4pVCKd4wgoHI4pjknBRTLrzYp6AL_D-j6MjcQSvQ/edit

**Comment by kkrysiak** [comment]
From discussions with members of the Melbourne Genomics Health Alliance, this is a priority. Partially because we don't have a nice way to dig through variants that we do have. Aliases only work on the quick search, browse and advanced search pages. On a gene page, unless you are searching for an exact variant, you can't get there.

**Comment by kkrysiak** [comment]
Everyone loves the COSMIC way of filtering...there is a histogram of events. That sort of thing would work for us as well. I love the lolliplot idea but I don't know about implementation. Shahil's visual (lolliplot that replaced variant occurrences with evidence count) was awesome for this sort of thing. But lolliplots don't really work that well for gene-level events.

**Comment by malachig** [comment]
We have talked about the idea of collaborating with the ProteinPaint team at St. Judes to resolve this issue.  Its not done yet, but it could be a way to resolve this...
